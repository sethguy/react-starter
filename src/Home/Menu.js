import React, { Component } from 'react';
export class Menu extends Component {
  node;
  componentDidMount() {
    document.body.addEventListener('mousedown', this.onAnyClick);
  }
  onAnyClick = (event) => {
    if (this.node && this.node.contains(event.target)) {
      return;
    }
    this.closeContextMenu();
  };

  closeContextMenu = () => {
    this.props.closeContextMenu()
  };
  render() {
    return (<div ref={node => this.node = node} style={{
      top: `${this.props.y}px`,
      left: `${this.props.x}px`,
    }} className="context-menu border"> faske </div>);
  }
}
