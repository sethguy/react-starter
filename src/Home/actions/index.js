export const loadEvents = (params) => async(dispatch, getState) => {
    dispatch({type: "loadEvent", params})
    const results = await getEvents(params)
    dispatch({type: "onEventData", ...results})
}

const getEvents = async(params) => {
    return Promise.resolve({eventData: {id:'fasjlkhkl6',name:'party time !'}})
}
