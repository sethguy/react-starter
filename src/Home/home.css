.context-menu-position-wrap {
  position: absolute;
}

.context-menu {
  position: relative;
  width: 250px;
  background-color: white;
}

.context-menu-item {
  position: relative;
  min-height: 20px;
  margin: 5px 0;
  overflow: visible;
}

.sub-context-menu {
  position: relative;
  width: 250px;
}

.sub-context-menu-abs {
  position: absolute;
}