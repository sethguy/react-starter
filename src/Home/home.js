import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  'col-class': {
    'border': [{ 'unit': 'px', 'value': 1 }, { 'unit': 'string', 'value': 'solid' }, { 'unit': 'string', 'value': 'black' }],
    'margin': [{ 'unit': 'px', 'value': 0 }, { 'unit': 'px', 'value': 12.5 }, { 'unit': 'px', 'value': 0 }, { 'unit': 'px', 'value': 12.5 }]
  }
});
