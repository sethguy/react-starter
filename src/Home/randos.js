export const randos = ["bang",
  "satisfying",
  "plausible",
  "snobbish",
  "incompetent",
  "hope",
  "knock",
  "type",
  "draconian",
  "upset",
  "fluffy",
  "injure",
  "account",
  "label",
  "employ",
  "friction",
  "walk",
  "border",
  "activity",
  "park",
  "guttural",
  "throat",
  "crate",
  "deranged",
  "standing",
  "kettle",
  "wheel",
  "tent",
  "hall",
  "kneel",
  "shelf",
  "gifted",
  "key",
  "silk",
  "venomous",
  "spotless",
  "gullible",
  "sleepy",
  "wandering",
  "fear",
  "confuse",
  "amusing",
  "warn",
  "melt",
  "few",
  "north",
  "eight",
  "ducks",
  "defective",
  "command",
  "elfin",
  "magical",
  "sidewalk",
  "library",
  "found",
  "eminent",
  "lacking",
  "selfish",
  "fat",
  "nonchalant",
  "stew",
  "ring",
  "harbor",
  "start",
  "aboard",
  "psychotic",
  "languid",
  "scandalous",
  "scorch",
  "helpless",
  "experience",
  "mean",
  "unbecoming",
  "clear",
  "tendency",
  "reflect",
  "tightfisted",
  "slope",
  "calendar",
  "charming",
  "pancake",
  "extra-small",
  "alive",
  "economic",
  "full",
  "part",
  "borrow",
  "lyrical",
  "melodic",
  "silly",
  "rampant",
  "pail",
  "badge",
  "spade",
  "abortive",
  "girls",
  "art",
  "dirt",
  "violent",
  "carriage",
];


