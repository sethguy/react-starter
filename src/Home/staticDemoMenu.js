import React from 'react';
import { MenuItem } from './Menu';
const staticDemoMenu = () => {
  return (<MenuItem title="cell Types">
    <MenuItem title="cell Type 1">
      <MenuItem title="option 1"></MenuItem>
      <MenuItem title="option 2"></MenuItem>
    </MenuItem>
    <MenuItem title="cell Type 2"></MenuItem>
  </MenuItem>
    ,
    <MenuItem title="input Types">
      <MenuItem title=" input option 1"></MenuItem>
      <MenuItem title=" input option 2"></MenuItem>
    </MenuItem>
    ,
    <MenuItem title="input2 Types">
      <MenuItem title=" input2 option 1">
        <MenuItem title=" input3 option 1"></MenuItem>
        <MenuItem title=" input3 option 23"></MenuItem>
        <MenuItem title=" input3 option 13"></MenuItem>
        <MenuItem title=" input3 option 24"></MenuItem>

        <MenuItem title=" input3 option 155"></MenuItem>
        <MenuItem title=" input3 option 266"></MenuItem>

      </MenuItem>
      <MenuItem title=" input2 option 2"></MenuItem>
    </MenuItem>);
};
