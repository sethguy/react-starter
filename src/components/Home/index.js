import React, {Component} from 'react';
import {loadEvents} from '../../actions/events'
import {connect} from 'react-redux';

class Home extends Component {
    componentDidMount() {
        this
            .props
            .loadEvents({})
    }
    render() {
        return (
            <div className="h-100" onContextMenu={(event)=>{
			console.log("TCL: Home -> render -> event", event)
                
            }} >
                <div>
                    <p>
                        HOme
                    </p>
                </div>
                <p>
                    {this.props.eventData.name}
                </p>
            </div>

        );
    }
}

const defaultEvent = {};

const mapStateToProps = ({
    events
}, ownProps) => ({
    eventData: events.eventData || defaultEvent
})

export default connect(mapStateToProps, {loadEvents})(Home)
