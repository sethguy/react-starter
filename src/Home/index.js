import React, { Component } from 'react';
import { loadEvents } from './actions'
import { connect } from 'react-redux';
import './home.css'
import { Menu, MenuItem } from './Menu';
import { randos } from './randos'


const getRandoConbo = () => {

    const r1 = Math.floor(Math.random() * randos.length - 1)
    const r2 = Math.floor(Math.random() * randos.length - 1)

    const wordOne = randos[r1]
    const wordtwo = randos[r2]

    const word = `${wordOne}-${wordtwo}`
    return word;
}
class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            contextMenuPosition: { x: 0, y: 0 }
        }

    }
    componentDidMount() {
        this.buildDemoList();
    }
    getMenuItems = () => {
        const blocks = new Array(5).fill("").map(space => {
            return {
                title: getRandoConbo(),
                subs: []
            }
        })
        return blocks;
    }
    listWithMenus = () => {
        return this.getMenuItems().map(menuitem => {
            return {
                ...menuitem,
                subs: this.getMenuItems()
            }
        })
    }

    buildDemoList = () => {

        const toplist = this.getMenuItems().map(menuitem => {

            return {
                ...menuitem,
                subs: this.getMenuItems().map(menuitem => {
                    return {
                        ...menuitem,
                        subs: this.getMenuItems().map(menuitem => {
                            return {
                                ...menuitem,
                                subs: this.getMenuItems().map(menuitem => {
                                    return {
                                        ...menuitem,
                                        subs: this.getMenuItems()
                                    }
                                })
                            }
                        })
                    }
                })
            }
        });
        this.setState({
            menulist: toplist
        })
    }
    openContextMenu = (event) => {
        this.setState({
            showContextMenu: true,
            contextMenuPosition: { x: event.x || 0, y: event.y || 0 }
        })
    }
    onCloseContextMenu = () => {
        this.setState({
            showContextMenu: false,
            contextMenuPosition: { x: 0, y: 0 }
        });
    };
    onContextMenuClick = (event) => {
        event.preventDefault()
        // nice trick //const positions = Object.keys(event).filter(key => key.indexOf('Y') > -1 || key.indexOf('X') > -1)
        const { screenX, screenY, clientX, clientY, pageX, pageY, movementX, movementY } = event
        this.openContextMenu({
            x: clientX,
            y: clientY,
        })
    }
    render() {
        const { x, y } = this.state.contextMenuPosition;
        return (
            <div className="h-100" style={{ backgroundColor: 'black' }} onContextMenu={this.onContextMenuClick} >
                {this.state.showContextMenu && <Menu {...{ x, y, closeContextMenu: this.onCloseContextMenu }} >

                    {this.state.menulist && this.state.menulist.map(list => {
                        return (
                            <MenuItem title={list.title} >
                                {list.subs && list.subs.map(sublist => {
                                    return (
                                        <MenuItem title={sublist.title} >
                                            {sublist.subs && sublist.subs.map(subsublist => {
                                                return (
                                                    <MenuItem title={subsublist.title} >
                                                        {subsublist.subs && subsublist.subs.map(subsubsublist => {
                                                            return (
                                                                <MenuItem title={subsubsublist.title} >
                                                                </MenuItem>
                                                            )
                                                        })}
                                                    </MenuItem>
                                                )
                                            })}
                                        </MenuItem>
                                    )
                                })}
                            </MenuItem>
                        )
                    })}

                </Menu>}
            </div>
        );
    }
}

const defaultEvent = {};

const mapStateToProps = ({
    events
}, ownProps) => ({
    eventData: events.eventData || defaultEvent
})

export default connect(mapStateToProps, { loadEvents })(Home)
