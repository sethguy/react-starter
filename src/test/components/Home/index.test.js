import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Home from '../../../components/Home'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares)

const ProviderWrapper = ({ store }) => {

  return (
    <Provider store={store}>
      <Home />
    </Provider>
  )
}

describe("Home", () => {
  test("it matches the snapshot", () => {
    const props = {}
    const renderer = new ShallowRenderer()
    const result = renderer.render(<ProviderWrapper store={mockStore({})} />)
    expect(result).toMatchSnapshot()
  });
});