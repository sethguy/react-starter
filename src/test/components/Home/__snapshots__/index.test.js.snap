// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home it matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Home) />
</Provider>
`;
